assets/src/stories-editor/blocks/amp-story-cta/edit.js

```diff
@@ -8,128 +8,97 @@ import PropTypes from 'prop-types';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Component } from '@wordpress/element';
-import {
-	Dashicon,
-	IconButton,
-} from '@wordpress/components';
-import {
-	URLInput,
-	RichText,
-} from '@wordpress/block-editor';
+import { useState, useEffect } from '@wordpress/element';
+import { Dashicon, IconButton } from '@wordpress/components';
+import { URLInput, RichText } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import './edit.css';
-import { select } from '@wordpress/data';
 import { getUniqueId, setInputSelectionToEnd } from '../../helpers';
 import { getBackgroundColorWithOpacity } from '../../../common/helpers';
 import { DraggableText } from '../../components';
 
-class CallToActionEdit extends Component {
-	constructor( props ) {
-		super( props );
-
-		if ( ! props.attributes.anchor ) {
-			this.props.setAttributes( { anchor: getUniqueId() } );
-		}
-
-		this.state = {
-			isEditing: false,
-			hasOverlay: true,
-		};
-
-		this.nodeRef = null;
-	}
-
-	bindRef = ( node ) => {
-		if ( ! node ) {
-			return;
-		}
-		this.nodeRef = node;
-	}
-
-	toggleIsEditing = ( enable ) => {
-		if ( enable !== this.state.isEditing ) {
-			this.setState( {
-				isEditing: ! this.state.isEditing,
-			} );
-		}
-	}
-
-	toggleOverlay = ( add ) => {
-		if ( add !== this.state.hasOverlay ) {
-			this.setState( {
-				hasOverlay: ! this.state.hasOverlay,
-			} );
+const CallToActionEdit = ( {
+	attributes,
+	backgroundColor,
+	className,
+	clientId,
+	fontSize,
+	isSelected,
+	name,
+	setAttributes,
+	textColor,
+} ) => {
+	const {
+		anchor,
+		text,
+		url,
+		customBackgroundColor,
+		opacity,
+		btnPositionTop,
+		btnPositionLeft,
+	} = attributes;
+
+	const [ isEditing, setIsEditing ] = useState( false );
+	const [ hasOverlay, setHasOverlay ] = useState( true );
+
+	useEffect( () => {
+		if ( ! anchor ) {
+			setAttributes( { anchor: getUniqueId() } );
 		}
-	}
+	}, [ anchor, setAttributes ] );
 
-	componentDidUpdate( prevProps, prevState ) {
-		const { isSelected } = this.props;
+	useEffect( () => {
 		// If the block was unselected, make sure that it's not editing anymore.
-		if ( ! isSelected && prevProps.isSelected ) {
-			this.toggleIsEditing( false );
-			this.toggleOverlay( true );
+		if ( ! isSelected ) {
+			setIsEditing( false );
+			setHasOverlay( true );
 		}
-		if ( this.state.isEditing && ! prevState.isEditing ) {
+	}, [ isSelected ] );
+
+	useEffect( () => {
+		if ( isEditing ) {
 			setInputSelectionToEnd( '.is-selected .amp-block-story-cta__link' );
 		}
-	}
+	}, [ isEditing ] );
```

Comment:
Curious if there's a reason for having a separate `useEffect` for `isEditing` and `isSelecting` instead of passing `[ isEditing, isSelected ]` together.

Reply:
Do you mean because there are `isEditing` and `isSelected` checks within the effects anyway?

IMO it's a bit more explicit this way, so they only really fire when necessary.

Reply:
Yep, it makes more sense here probably anyway since `isSelected` change might trigger `isEditing` change etc.